lib/spack/spack/compiler.py

```diff
@@ -117,6 +117,20 @@ def check(exe):
     def version(self):
         return self.spec.version
 
+    # Default implementation of OpenMP linking flag.
+    # Override in derived classes if needed
+    @property
+    def openmp_flag(self):
+        return "-fopenmp"
+
+
+    # Default implementation of c++11 linking flag.
+    # Override in derived classes if needed
+    @property
+    def cxx11_flag(self):
+        return "-std=c++11"
```

Comment:
The more I think about it, these properties should probably look something like:
```python
# This property should be overridden in the compiler subclass if
# OpenMP is supported by that compiler
@property
def openmp_flag(self):
    # If it is not overridden, assume it is not supported and warn the user
    tty.die("The compiler you have chosen does not currently support OpenMP. If you think it should, please edit the compiler subclass and submit a pull request or issue.")


# This property should be overridden in the compiler subclass if
# C++11 is supported by that compiler
@property
def cxx11_flag(self):
    # If it is not overridden, assume it is not supported and warn the user
    tty.die("The compiler you have chosen does not currently support C++11. If you think it should, please edit the compiler subclass and submit a pull request or issue.")
```
The benefit of this is that if someone adds a new compiler but forgets to override these properties, or the compiler does not support them, Spack warns the user. Otherwise, the current implementation will just use the wrong flag and crash. Feel free to shorten up the message if you want.